### ISSUE TITLE
Please disable or fix waterway crossing highway warnings, many are in contradiction with wiki and established practice

### ISSUE THREAD

**Issue by RicoZ-OSM** [keepright/keepright#60]
I believe this one http://www.openstreetmap.org/browse/way/23391462, and every single of the waterway crossing highway concerning the nearby dam (eg http://www.openstreetmap.org/browse/way/25298372, http://www.openstreetmap.org/browse/way/25298372 ) are false positives.

The first one is perhaps caused by assuming default layer=-1 for tunnels which is wrong.

The highway crossing dam is not ultimately settled in the wiki. You may participate in the discussion ( http://wiki.openstreetmap.org/wiki/Talk:Tag:waterway%3Ddam ) but until this is settled please don't flag things which are not widely agreed to be wrong.

In particular following should be considered no-warn until there is a clear consensus otherwise:
* highway crossing dam, weir and similar *with* or *without* shared nodes
* highway sharing a single way with a dam,weir and similar which may also have a shared node (ford or not) with the waterway running though it

**Comment by mueschel** [comment]
> and every single of the waterway crossing highway concerning the nearby dam (eg http://www.openstreetmap.org/browse/way/25298372, http://www.openstreetmap.org/browse/way/25298372 ) are false positives.
Yes, they are. waterway=dam was not in the exclusion list for waterways that are no waterways. Will be in one of my next commits.


> I believe this one http://www.openstreetmap.org/browse/way/23391462, [is] false positive
This tagging can't be right. Currently the tunnel is on the same layer as the forest above it. Either the forest needs a layer=+1 (which doesn't make any sense) or the tunnel needs -1 or -2, depending on whether it is above or below the waterways.

**Comment by RicoZ-OSM** [comment]
Please also the weirs, not sure if there are more exceptions.

You may argue that the tunnel tagging is not correct but the error message is 100% incorrect, it would be good  to fix the specific cause for this before looking what else might or not be wrongly mapped here. 

Considering your objection, you can by extension argue that it is wrong for any tunnel or bridge not to have a layer!=0 because there is always natural ground bellow/above it whether or not there is an actual forest mapped. I would agree with you and wiki recommends it but many people won't agree and considering the huge number of bridges/tunnels without layer there are probably more important warnings to displayed.

**Comment by mueschel** [comment]
Weirs were already included.

You actually just gave the argument why a layer of +-1 is assumed for tunnels and bridges: There are so many without this tag. 

The discussion on this topic is popping up every now and then, and the common opinion usually is that there is no default and it is up to the data consumer to assume what it wants. Currently 23% of bridges do not have a layer tag and I am convinced that in 99% of these cases layer=1 is the correct assumption. Assuming the default layer 0 would lead to an awful lot of additional errors just due to that.

**Comment by RicoZ-OSM** [comment]
If I understand it correctly you are worried about false positives if you would assume default layer=0 for bridges/tunnels? 
That would seem to be not very hard to get right, just assume  +/-0.5 as default layer for bridge/tunnel. Seems to be similar to what many other data consumers do.

When you are worried about false positives did you consider that level and location also complicate the picture? The very least and easiest to avoid false positives is to drop checks for objects that have either of level or  location==underground  or location==overhead)

**Comment by RicoZ-OSM** [comment]
ping - did you think about using layer=+/- 0.5 for bridges/tunnles missing an explicit layer?
